src/inject/dynamic-theme/adopted-style-manger.ts

```diff
@@ -0,0 +1,102 @@
+import {Theme} from '../../definitions';
+import {createStyleSheetModifier} from './stylesheet-modifier';
+import {forEach} from '../../utils/array';
+
+let adoptedStyleOverrides = new WeakMap<CSSStyleSheet, CSSStyleSheet>();
+let adoptedStyleOwners = [] as Array<ShadowRoot | Document>;
+let overrideList = new WeakSet<CSSStyleSheet>();
+
+export function removeAdoptedStyleSheets() {
+    forEach(adoptedStyleOwners, (node) => {
+        forEach(node.adoptedStyleSheets, (adoptedStyleSheet) => {
+            if (overrideList.has(adoptedStyleSheet)) {
+                const newSheets = [...node.adoptedStyleSheets];
+                const existingIndex = newSheets.indexOf(adoptedStyleOverrides.get(adoptedStyleSheet));
+                if (existingIndex >= 0) {
+                    newSheets.splice(existingIndex, 1);
+                }
+                node.adoptedStyleSheets = newSheets;
+            }
+        });
+    });
+    cleanAdoptedStyleSheets();
+}
+function cleanAdoptedStyleSheets() {
```

Comment:
This should become exported function and called in `index.ts` file when necessary.